src/field/vecops.cxx

```diff
@@ -186,8 +186,17 @@ const Field3D Div(const Vector3D& v, CELL_LOC outloc, const std::string& method)
   Vector3D vcn = v;
   vcn.toContravariant();
 
-  Field3D result = DDX(vcn.x.getCoordinates()->J * vcn.x, outloc, method);
-  result += DDY(vcn.y.getCoordinates()->J * vcn.y, outloc, method);
+  Field3D result = 0.0;
+  
+  if(!vcn.y.hasParallelSlices()){
+    Field3D vcnJy = vcn.y.getCoordinates()->J * vcn.y;
+    localmesh->communicate(vcnJy);
+    result = DDY(vcnJy, outloc, method);
+  }else{
+    result = DDY(vcn.y.getCoordinates()->J * vcn.y, outloc, method);
+  }
```

Comment:
This sort of initialisation from a lambda is useful when there's some logic beyond a ternary required to initialise a variable that you don't want to wrap up into a standalone function.

Having no separation between declaration and initialisation is really useful to prevent bugs like use before initialisation. Even if the code is correct when its written, someone could come along and insert new code between the declaration and initialisation. That's impossible if it's done immediately

Reply:
But you can initalise it like this:
```
   auto vcnJy = vcn.y.getCoordinates()->J * vcn.y;
   if (not vcn.y.hasParallelSlices()) {
     localmesh->communicate(vcnJy);
   }
   auto result = DDY(vcnJy, outloc, method);
```
This has the disadvantage that the scope of `vcnJy` is larger then it could be, but it is imho easier to understand then the lambda.